src/util/BaseRestClient.ts

```diff
@@ -199,7 +201,7 @@ export default abstract class BaseRestClient {
       throw new Error('Private endpoints require api and private keys set');
     }
 
-    if (this.timeOffset === null) {
+    if (this.timeOffset === null || this.options.syncTimeBeforePrivateRequests) {
       await this.syncTime();
     }
```

Comment:
On reviewing this again, take a look at the `syncTime` function further down (line 450):
```typescript
    if (!force && !this.options.enable_time_sync) {
      this.timeOffset = 0;
      return Promise.resolve(false);
    }
```

It will bypass time sync if enable_time_sync is still disabled, although you could force it by changing 205 to pass the boolean:
```typescript
    if (this.timeOffset === null || this.options.syncTimeBeforePrivateRequests) {
      await this.syncTime(!!this.options.syncTimeBeforePrivateRequests);
    }
```

Reply:
updated.

In addition, do I need to add instructions in readme

Reply:
Hmm your description sounds to me like any adjustment of the offset is incorrect and causes issues because of one (out of many) bad request. So with this turned on (including the force:true in sync time()), you're seeing more frequent time errors? 

Because without that change (force:true), syncTime won't function unless enable_time_sync is true...could you do more testing to see if this feature makes sense & improves things?

Reply:
> Because without that change (force:true), syncTime won't function unless enable_time_sync is true...could you do more testing to see if this feature makes sense & improves things?

I added a log in BaseRestClient.js
```
      console.log("before:", this.timeOffset);
      // Build a request and handle signature process
      const options = yield this.buildRequest(
        method,
        requestUrl,
        params,
        isPublicApi
      );
      console.log("after: ", this.timeOffset);
```
And still the manually change in `prepareSignParams`:
```
  prepareSignParams(method, signMethod, params, isPublicApi) {
    return __awaiter(this, void 0, void 0, function* () {
      if (isPublicApi) {
        return {
          originalParams: params,
          paramsWithSign: params,
        };
      }
      if (!this.key || !this.secret) {
        throw new Error("Private endpoints require api and private keys set");
      }
      //if (this.timeOffset === null) {
      yield this.syncTime();
      //}
      return this.signRequest(params || {}, method, signMethod);
    });
  }
```
It print out 
```
before: null
after:  0
```
So it turns out I don't need an offset for my local time

Reply:
I have a mac and my friend have a windows 11
Time errors occur frequently in wsl of my friend's win, while windows and mac do not currently see frequent problems. However, if the bybit server has time issues, I need to force synchronization.